sound/soc/sof/ipc4-topology.c

```diff
@@ -1464,6 +1493,10 @@ sof_ipc4_prepare_copier_module(struct snd_sof_widget *swidget,
 				copier_data->gtw_cfg.node_id |= SOF_IPC4_NODE_INDEX(group_id);
 			}
 		}
+
+		if (hw_param_mask(pipeline_params, SNDRV_PCM_HW_PARAM_FORMAT) !=
+			hw_param_mask(fe_params, SNDRV_PCM_HW_PARAM_FORMAT))
+			sof_ipc4_apply_fe_params_to_dai(swidget, fe_params, copier_data, ipc4_copier);
```

Comment:
how would this work for a mixer use case, where there might be two or more FEs connected to a DAI?

I really don't see how this might work.

Reply:
It will use the params of the  first stream since the dai is not prepared again for other streams in mixer case.

Reply:
that's broken. Or rather just as broken as IPC3. That will result in different DAI configurations depending on the format of the first stream. This is really not good at all. The mixer needs to have its own configuration and we absolutely do not want to propagate hw_params across pipelines.

Reply:
@plbossart  so do we need to first refine the hw_params across pipelines ? then check BE fixup ?  Do you mean that DAI format depend on mixer ? Current mixer is 32bit then my pr is unnecessary.  Or only apply my pr to pipeline without mixer ? Thanks!

Reply:
@RanderWang I must admit not understanding what you are trying to fix or improve If the mixer is required by firmware to work on 32 bits, then how would we make the decision that the DAI needs to work with less than 32 bits? we would need to have some sort of guidance, maybe from topology, to define what the hardware needs or optimize bandwidth requirements.
The irony is that the DAI capabilities are ONLY exposed on the SSP case, for HDAudio and SoundWire we have zero information.

Reply:
Thanks!